HomeAutomationServer/Models/AppCache.cs

```diff
@@ -17,27 +17,32 @@
 using Amazon.S3.Model;
 using Amazon.SimpleNotificationService;
 using Amazon.SimpleNotificationService.Model;
+using api;
 
 namespace HomeAutomationServer.Models
 {
-    static public class AppCache  // A cache to temporarily store app information while waiting on 
-    {                      // information request.
+	static public class AppCache  // A cache to temporarily store app information while waiting on 
+	{                      // information request.
 
-        // A JSON array of device blobs
-        private static JArray deviceBlobs = new JArray();
-        // add any other JArrays containing blobs here
-        
-        ////////////////////////////////////////////////////////////////////////////////////////
-        //
-        // deviceBlobs methods
+		// A JSON array of device blobs
+		private static Dictionary<string, Device> deviceBlobs = new Dictionary<string, Device>();
+		// add any other JArrays containing blobs here
+
+		////////////////////////////////////////////////////////////////////////////////////////
+		//
+		// deviceBlobs methods
 
 		/*************** DEBUG MODE METHOD *************/
 
 		static public bool AddDeviceBlob_DEBUG(JObject blob)
 		{
-			deviceBlobs[(string)blob["deviceID"]] = blob;
+			string blob_string = blob.ToString ();
+			Device dev = Interfaces.DeserializeDevice(blob_string, null, null, new TimeFrame()); // to convert the JSON blob to an actual Device object.
```

Comment:
For the app, you should be using the CreateDevice() function, which doesn't take the two null arguments, and will create this such that it can talk to the server.